app/src/lib/git/status.ts

```diff
@@ -217,12 +221,15 @@ export async function getStatus(
 
   const workingDirectory = WorkingDirectoryStatus.fromFiles([...files.values()])
 
+  const mergeHeadFound = await isMergeHeadSet(repository)
+
   return {
     currentBranch,
     currentTip,
     currentUpstreamBranch,
     branchAheadBehind,
     exists: true,
+    mergeHeadFound,
```

Comment:
The biggest change about this PR is that this becomes the source of truth for detecting whether a repository is in a conflicted state. We can't rely on `AppFileStatus.Conflicted` or `AppFileStatus.Resolved` for situations when a binary merge conflict is resolved, because that changes the state of the index to remove the conflicts.

This has flow-on effects in some places where we do checks like this:

```
    const workingDirectioryHasConflicts = status.workingDirectory.files.some(
      file =>
        file.status === AppFileStatus.Conflicted ||
        file.status === AppFileStatus.Resolved
    )
```

I plan to extract this change to it's own PR in the morning so we can discuss these details separate to this PR.

Reply:
See #6116 for this PR